tests/tools/unit/test_tools.py

```diff
@@ -0,0 +1,191 @@
+import dataclasses
+
+import pytest
+
+from tools import (calibrate_encoder, comms, control_motor, read_sensor,
+                   update_calibration, upload_bootloader, upload_firmware,
+                   view_control_loop)
+
+
+@dataclasses.dataclass
+class BaseArgs:
+    """Default arguments used by all tools."""
+    serial: str = '/dev/ttyUSB0'
+    board_ids: str = '1'
+    baud_rate: int = comms.COMM_DEFAULT_BAUD_RATE
+
+
+def _default_mock_comms(mocker):
```

Comment:
rather than call `_default_mock_comms` in every function, you can create a fixture, then use that fixture everywhere as input.
ie
```suggestion
@pytest.fixture
def default_mock_comms(mocker):
```
then use `default_mock_comms` as the function input to all the tests instead of `mocker`